tools/ctl/ctl.go

```diff
@@ -1184,9 +1185,91 @@ var runAgent = &cobra.Command{
 		return nil
 	},
 }
+
+var localAgent = &cobra.Command{
+	Use:   "local-agent --project <project> --agent-api <URI> --metadata-bucket <bucket> --ecosystem <ecosystem> --package <name> --version <version> --artifact <name>",
+	Short: "Run benchmark",
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := cmd.Context()
+		t := rebuild.Target{Ecosystem: rebuild.Ecosystem(*ecosystem), Package: *pkg, Version: *version, Artifact: *artifact}
+		if t.Ecosystem == "" || t.Package == "" || t.Version == "" || t.Artifact == "" {
+			log.Fatal("ecosystem, package, version, and artifact must be provided")
+		}
+		if *project == "" {
+			return errors.New("project must be provided")
+		}
+		if *agentApiUri == "" {
+			return errors.New("agent API endpoint not provided")
+		}
+		if *metadataBucket == "" {
+			return errors.New("metadata bucket not provided")
+		}
+		fire, err := firestore.NewClient(cmd.Context(), *project)
+		if err != nil {
+			return errors.Wrap(err, "creating firestore client")
+		}
+		sessionTime := time.Now().UTC()
+		sessionID := fmt.Sprintf("session-%s", sessionTime.Format(time.RFC3339))
```

Comment:
i'd be fine with including something indicating this was local. there's no purpose in my mind to putting "session" in the session key, though.